[documentation] added iOS Shortcut Link for Uploading PDF to paperless

dabenzel opened the pull request:
<!--
Note: All PRs with code changes should be targeted to the `dev` branch, pure documentation changes can target `main`
-->

## Proposed change

<!--
Please include a summary of the change and which issue is fixed (if any) and any relevant motivation / context. List any dependencies that are required for this change. If appropriate, please include an explanation of how your proposed change can be tested. Screenshots and / or videos can also be helpful if appropriate.
-->

Fixes # (issue)

## Type of change

<!--
What type of change does your PR introduce to Paperless-ngx?
NOTE: Please check only one box!
-->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (please explain)

Documentation change

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

dabenzel committed:
I have created a shortcut to Upload PDF files

//

qcasey commented:
Thank you for your contribution! We'll need someone with an iOS device to test that this works as expected.

shamoon commented:
This is a cool idea but unfortunately its not working for me. I just get
`[2022-05-05 11:58:55,186] [WARNING] [django.request] Unsupported Media Type: /api/documents/post_document/` and the document never uploads. Ive tried with a few PDFs. If its really rock-solid (or you can get it to be so) this would be great to include

dabenzel commented:
Thats interesting, because I've set the shortcut to only accept PDF files... I've tested and created mine with the app `QuickScan` and from there they work.

Is the second line "PDF Document" in the share dialog?

shamoon commented:
Yea it’s a pdf doc which uploads fine other ways https://filesamples.com/samples/document/pdf/sample3.pdf

![0AA3A534-4FA6-40F0-9875-BB2E210D89EF](https://user-images.githubusercontent.com/4887959/167010890-9790eaef-12cb-425d-a028-06fdd50d7a59.png)

dabenzel commented:
Hi! I can reprouce that. If you save the file on the device first, the shortcut works

dabenzel commented:
Made a few more tests. Seems like iOS really wants a local file. It seems that sharing from a website-pdf doesnt work.

shamoon commented:
Ah cool it does work with a local file. This seems pretty useful!

If you update the shortcut does the URL here have to change? Im not super familiar with publicly-shared shortcuts.

dabenzel commented:
My shortcut currently looks like this

Can you look at yours?

This is the current version:
https://www.icloud.com/shortcuts/9a74a77110264f93ad49f8c69f0abbb1


shamoon ***@***.***> schrieb am Do. 5. Mai 2022 um 22:10:

> Ah cool it does work with a local file. This seems pretty useful!
>
> If you update the shortcut does the URL here have to change? Im not super
> familiar with publicly-shared shortcuts.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/paperless-ngx/paperless-ngx/pull/878#issuecomment-1119001875>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAS5Q22RNRCUXHSF57ODBVDVIQTK7ANCNFSM5VDDDGAQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

shamoon commented:
Sorry I just meant like if you / we make updates to the shortcut setup. Like I would update some of the text here slightly (not important just a little picky). I will leave as a code review.

Otherwise this is great and would be happy to merge.

shamoon commented on docs/scanners.rst:
```diff
@@ -105,6 +105,8 @@ You can use your phone to "scan" documents. The regular camera app will work, bu
 
 On Android, you can use these applications in combination with one of the :ref:`Paperless-ngx compatible apps <usage-mobile_upload>` to "Share" the documents produced by these scanner apps with paperless. On iOS, you can share the scanned documents via iOS-Sharing to other mail, WebDav or FTP apps.
 
+There is a Shortcut on iOS where you can directly share PDF Documents to paperless here: https://www.icloud.com/shortcuts/52ce6292ea4e4a01951e14454b88bb14
```
```suggestion
There is also an iOS Shortcut that allows you to directly upload PDF documents available here: https://www.icloud.com/shortcuts/52ce6292ea4e4a01951e14454b88bb14
Please note this only works for documents downloaded to iCloud / the device, in other words not directly from a URL.
```

shamoon commented on docs/scanners.rst:
```diff
@@ -105,6 +105,8 @@ You can use your phone to "scan" documents. The regular camera app will work, bu
 
 On Android, you can use these applications in combination with one of the :ref:`Paperless-ngx compatible apps <usage-mobile_upload>` to "Share" the documents produced by these scanner apps with paperless. On iOS, you can share the scanned documents via iOS-Sharing to other mail, WebDav or FTP apps.
 
+There is a Shortcut on iOS where you can directly share PDF Documents to paperless here: https://www.icloud.com/shortcuts/52ce6292ea4e4a01951e14454b88bb14
```
Please update the setup text of the shortcut to something like below (just grammar, capitalization etc nothing major)
`Please enter the Url (with http) for Paperless. No trailing slash` --> `Please enter the URL (with https) for your Paperless install. Do not include trailing slash.`

`Please enter your API-Key (from Admin-Panel)` --> `Please enter Auth token (found in Admin section).`

dabenzel replied:
Hi!
I'll update it tomorrow. I think after saving it will generate  a new Link

dabenzel committed:
Update docs/scanners.rst

Co-authored-by: shamoon <4887959+shamoon@users.noreply.github.com>

shamoon commented:
FYI, still awaiting an updated version of the shortcut from OP

shamoon commented on docs/scanners.rst:
```diff
@@ -105,6 +105,9 @@ You can use your phone to "scan" documents. The regular camera app will work, bu
 
 On Android, you can use these applications in combination with one of the :ref:`Paperless-ngx compatible apps <usage-mobile_upload>` to "Share" the documents produced by these scanner apps with paperless. On iOS, you can share the scanned documents via iOS-Sharing to other mail, WebDav or FTP apps.
 
+There is also an iOS Shortcut that allows you to directly upload PDF documents available here: https://www.icloud.com/shortcuts/52ce6292ea4e4a01951e14454b88bb14
```
```suggestion
There is also an iOS Shortcut that allows you to directly upload PDF documents available here: https://www.icloud.com/shortcuts/364d60428ac34979b8d767cf80f768e6
```

shamoon reviewed (changes requested):
I updated with a new version that fixes the text and also makes it available as a "Quick Action" in the Finder in macOS. Let me know if youre OK with these changes @dabenzel

dabenzel commented:
Hi!
I also added "show <content from url>" at the end. Might make it able to
enable/disable it at the weekend. Then I'm at my mac and can do it easier.
If it has time until then, it would be fine.

Will also implement your changes then.

Am Mi., 11. Mai 2022 um 07:39 Uhr schrieb shamoon ***@***.***
>:

> ***@***.**** requested changes on this pull request.
>
> I updated with a new version that fixes the text and also makes it
> available as a "Quick Action" in the Finder in macOS. Let me know if youre
> OK with these changes @dabenzel <https://github.com/dabenzel>
> ------------------------------
>
> In docs/scanners.rst
> <https://github.com/paperless-ngx/paperless-ngx/pull/878#discussion_r869887750>
> :
>
> > @@ -105,6 +105,9 @@ You can use your phone to "scan" documents. The regular camera app will work, bu
>
>
>
>  On Android, you can use these applications in combination with one of the :ref:`Paperless-ngx compatible apps <usage-mobile_upload>` to "Share" the documents produced by these scanner apps with paperless. On iOS, you can share the scanned documents via iOS-Sharing to other mail, WebDav or FTP apps.
>
>
>
> +There is also an iOS Shortcut that allows you to directly upload PDF documents available here: https://www.icloud.com/shortcuts/52ce6292ea4e4a01951e14454b88bb14
>
> ⬇️ Suggested change
>
> -There is also an iOS Shortcut that allows you to directly upload PDF documents available here: https://www.icloud.com/shortcuts/52ce6292ea4e4a01951e14454b88bb14
>
> +There is also an iOS Shortcut that allows you to directly upload PDF documents available here: https://www.icloud.com/shortcuts/364d60428ac34979b8d767cf80f768e6
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/paperless-ngx/paperless-ngx/pull/878#pullrequestreview-968700993>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAS5Q2566IE3XHQS42RHAFTVJNBZJANCNFSM5VDDDGAQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

dabenzel commented:
hi @shamoon - can you test this one?
https://www.icloud.com/shortcuts/7696acad005443999cd6b1eaed3e6eba

dabenzel committed:
Update scanners.rst

shamoon commented:
Still works great, but why not just remove the option to show a confirmation and show the confirmation? I can’t imagine anyone being unhappy by getting a confirmation. I’d suggest keep it simple, just the URL and the auth token and show the confirmation.

tooomm commented:
>Made a few more tests. Seems like iOS really wants a local file. It seems that sharing from a website-pdf doesnt work.

Does this work when trying to upload a file via the share dialog from e.g. Dropbox ("Export file"), or a pdf attachment received in the Mail app?

I can not test this myself right now, but trying to understand the limitations of the iOS shortcut.

dabenzel commented:
Made a few tests.
OneDrive app didn’t work.
OneDrive via files app works
Mail pdf attachment works



tooomm ***@***.***> schrieb am Mo. 16. Mai 2022 um 11:19:

> Made a few more tests. Seems like iOS really wants a local file. It seems
> that sharing from a website-pdf doesnt work.
>
> Does this work when trying to upload a file via the share dialog from e.g.
> Dropbox ("Export file"), or a pdf attachment received in the Mail app?
>
> I can not test this myself right now, but trying to understand the
> limitations of the iOS shortcut.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/paperless-ngx/paperless-ngx/pull/878#issuecomment-1127431419>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAS5Q254NKCFJXGIXORSBJTVKIHKNANCNFSM5VDDDGAQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

dabenzel commented:
Nevermind. OneDrive app also worked directly via ios share dialogue

benzel ***@***.***> schrieb am Mo. 16. Mai 2022 um 12:33:

> Made a few tests.
> OneDrive app didn’t work.
> OneDrive via files app works
> Mail pdf attachment works
>
>
>
> tooomm ***@***.***> schrieb am Mo. 16. Mai 2022 um 11:19:
>
>> Made a few more tests. Seems like iOS really wants a local file. It seems
>> that sharing from a website-pdf doesnt work.
>>
>> Does this work when trying to upload a file via the share dialog from
>> e.g. Dropbox ("Export file"), or a pdf attachment received in the Mail app?
>>
>> I can not test this myself right now, but trying to understand the
>> limitations of the iOS shortcut.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/paperless-ngx/paperless-ngx/pull/878#issuecomment-1127431419>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAS5Q254NKCFJXGIXORSBJTVKIHKNANCNFSM5VDDDGAQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

tooomm commented:
>Made a few tests.
OneDrive via files app works
OneDrive app also worked directly via ios share dialogue
Mail pdf attachment works

Thanks for testing @dabenzel. That's great and should help with many workflows then! 👍 

The Paperless readme lists `PDF documents, images, plain text files` as natively supported.
Any particular reason to limit the shortcut to PDFs only?

shamoon commented on docs/scanners.rst:
```diff
@@ -105,6 +105,9 @@ You can use your phone to "scan" documents. The regular camera app will work, bu
 
 On Android, you can use these applications in combination with one of the :ref:`Paperless-ngx compatible apps <usage-mobile_upload>` to "Share" the documents produced by these scanner apps with paperless. On iOS, you can share the scanned documents via iOS-Sharing to other mail, WebDav or FTP apps.
 
+There is also an iOS Shortcut that allows you to directly upload PDF documents available here: https://www.icloud.com/shortcuts/7696acad005443999cd6b1eaed3e6eba
```
```suggestion
There is also an iOS Shortcut that allows you to directly upload text, PDF and image documents available here: https://www.icloud.com/shortcuts/d6dbf8dc1e454f8ea03b4c3d70058847
```

shamoon reviewed (changes requested):
Cool idea, this works well! I tested with plaintext docs and images. I also updated the shortcut to just show the check if it gets "OK" or a notification otherwise.

IMHO with these changes this is good to go if youre OK with it @dabenzel

dabenzel committed:
Update docs/scanners.rst

Co-authored-by: shamoon <4887959+shamoon@users.noreply.github.com>

dabenzel commented:
@shamoon you may need to remove step 2 as it creates a pdf of the input file

shamoon commented:
> @shamoon you may need to remove step 2 as it creates a pdf of the input file

Its a good question, I was going to leave it because it seems to work well and I think will ultimately support more files (basically if iOS can convert it it can be uploaded and Paperless will likely accept). But not sure if you feel thats an issue since it is changing the input file type.

tooomm commented:
I think the shortcut should probably not *silently* convert all files to PDF. People might prefer the original file.
That would be a feature on its own and the shortcut should state that: `Upload as PDF to Paperless`

Also, the generic document type "file" is selected as input over the dedicated "text" and "rich text" options.
But not sure if that's wanted/needed, or what the rich text one exactly is about.

I'm wondering if enabling `Safari web pages` or `URLs` as input would make e.g. PDF files in the browser work?

shamoon commented on docs/scanners.rst:
```diff
@@ -105,6 +105,9 @@ You can use your phone to "scan" documents. The regular camera app will work, bu
 
 On Android, you can use these applications in combination with one of the :ref:`Paperless-ngx compatible apps <usage-mobile_upload>` to "Share" the documents produced by these scanner apps with paperless. On iOS, you can share the scanned documents via iOS-Sharing to other mail, WebDav or FTP apps.
 
+There is also an iOS Shortcut that allows you to directly upload text, PDF and image documents available here: https://www.icloud.com/shortcuts/d6dbf8dc1e454f8ea03b4c3d70058847
+Please note this only works for documents downloaded to iCloud / the device, in other words not directly from a URL.
```
```suggestion
There is also an iOS Shortcut that allows you to directly upload text, PDF and image documents available here: https://www.icloud.com/shortcuts/d234abc0885040129d9d75fa45fe1154
Please note this only works for documents downloaded to iCloud / the device, in other words not directly from a URL.
```

shamoon committed:
Update shortcut link

shamoon commented:
Ok I updated to remove PDF conversion, still works great. Im pretty happy to have this tool, thanks @dabenzel!
